### PR TITLE
[UI] Mentions sidebar

### LINKED ISSUE #141: New feature : Mentions sidebar
Integrate new feature "show mentions in Explore view"

- [x] 3 mentions by labels (full list colapsed)

- [x] Active scroll

### PR DESCRIPTION
**Pull request by leiyre**
fix #141

### CODE REVIEW

File: frontend/pages/_id/index.vue [hunk 1]
```diff
@@ -23,7 +23,7 @@
     <div class="container">
       <div :class="['grid', annotationEnabled ? 'grid--editable' : '']">
         <Results :dataset="dataset" :headerHeight="headerHeight"> </Results>
-        <SideBar :dataset="dataset" />
+        <SideBar :dataset="dataset" :entities="dataset.entities" />
```
Comment (frascuchon) [inline frontend/pages/_id/index.vue]: we shouldn't include a task specific property here. For text classification there is no `entities` property defined and this should bring unexpected errors

File: frontend/components/commons/SideBar.vue [hunk 2]
```diff
@@ -5,101 +5,81 @@
...
-              <span class="records-number">{{ counter }}</span>
+        <div v-else>
+          <div class="sidebar__tabs">
+            <a href="#" :class="activeTab === 'mentions' ? 'active' : ''" @click.prevent="filteredMentionsBy('mentions')">Annotated as</a>
+            <a href="#" :class="activeTab === 'predicted_mentions' ? 'active' : ''" @click.prevent="filteredMentionsBy('predicted_mentions')">Predicted as</a>
+          </div>
+          <div class="scroll">
+            <div v-for="(prop, key) in filteredMentions" :key="key" :class="expandedMentionsGroup === key ? 'expanded' : ''">
+              <span :class="[`color_${entities.filter(e => e.text === key)[0].colorId}`, 'entity']">{{ key }}</span>
+              <SidebarCollapsableMentions :limit="expandedMentionsGroup && expandedMentionsGroup !== key ? 0 : currentMentionsLength" :entities="entities" :k="key" :object="filteredMentions" @limit="onShowMore(key)" />
```
Comment (frascuchon) [inline frontend/components/commons/SideBar.vue]: If the `SideBar` is a common component, we should hide specific task properties references (like mentions). Can we move the mentions logic into an specific sidebar component for token classification?

**Review by frascuchon — CHANGES_REQUESTED** [review]

File: frontend/components/commons/SideBar.vue (hunk 2 above)
Comment (leiyre) [reply frontend/components/commons/SideBar.vue]: @frascuchon I have separated global sidebar and two specific task based components

File: frontend/components/commons/SideBar.vue [hunk 3]
```diff
@@ -192,6 +73,9 @@ export default {
       this.openExportModal = false;
       this.exportAnnotations({ name: this.dataset.name });
     },
+    toggleVisibleMetrics() {
+      !this.visible ? (this.visible = true) : (this.visible = false);
```
Comment (frascuchon) [inline frontend/components/commons/SideBar.vue]: if `this.visible` if properly initialized, could we set as follows:
```suggestion
      this.visible  = !this.visible;
```

File: frontend/pages/_id/index.vue [hunk 4]
```diff
@@ -23,7 +23,10 @@
     <div class="container">
       <div :class="['grid', annotationEnabled ? 'grid--editable' : '']">
         <Results :dataset="dataset" :headerHeight="headerHeight"> </Results>
-        <SideBar :dataset="dataset" />
+        <SideBar :dataset="dataset" :annotationEnabled="annotationEnabled">
+          <TextClassificationMetrics v-if="dataset.task === 'TextClassification'" :dataset="dataset" />
+          <TokenClassificationMetrics v-else :dataset="dataset" :entities="dataset.entities" />
```
Comment (frascuchon) [inline frontend/pages/_id/index.vue]: Could we just pass dataset en refer to `dataset.entities` inside `TokenClassificationMetrics` component?

File: frontend/components/token-classifier/TokenClassificationMetrics.vue [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+  </div>
+</template>
+<script>
+export default {
+  props: {
+    dataset: {
+      type: Object,
+      required: true,
+    },
+    entities:{
```
Comment (frascuchon) [inline frontend/components/token-classifier/TokenClassificationMetrics.vue]: We can find entities inside `dataset` object

File: frontend/components/commons/SideBar.vue [hunk 6]
```diff
@@ -1,187 +1,68 @@
...
+import { mapActions } from "vuex";
 import "assets/icons/metrics";
 import "assets/icons/cross";
-import { mapActions } from "vuex";
-import { AnnotationProgress } from "@/models/AnnotationProgress";
-import { ObservationDataset } from "@/models/Dataset";
 export default {
-  // TODO clean and typify
   props: {
+    annotationEnabled: {
```
Comment (frascuchon) [inline frontend/components/commons/SideBar.vue]: compute this from `dataset`

**Review by frascuchon — CHANGES_REQUESTED** [review]

**Review by frascuchon — APPROVED** [review]

**Pull request merged**